fedmsg_meta_fedora_infrastructure/scm.py

```diff
@@ -56,15 +57,17 @@ def long_form(self, msg, **config):
             except KeyError:
                 repo = '.'.join(msg['topic'].split('.')[5:-1])
 
+            namespace = msg['msg']['commit'].get('namespace', 'rpms')
             rev = msg['msg']['commit']['rev']
 
             seen = msg['msg']['commit'].get('seen', False)
             if seen:
                 return already_seen_msg
 
-            url = 'https://src.fedoraproject.org/cgit/' + \
-                '{repo}.git/patch/?id={rev}'
-            response = requests.get(url.format(repo=repo, rev=rev))
+            url = 'https://src.fedoraproject.org/' + \
+                '{namespace}/{repo}/c/{rev}.patch'
+            response = requests.get(url.format(
```

Comment:
Block changed to:
~~~~
            url = 'https://src.fedoraproject.org/' + \
                '{namespace}/{repo}/c/{rev}.patch'
            url = url.format(namespace=namespace, repo=repo, rev=rev)

            try:
                response = requests.get(url, timeout=30)
                if response.status_code == 200:
                    return response.text
            except requests.exceptions.RequestException:
                return "Could not successfully call: {}".format(url)
~~~~